api/src/pcapi/connectors/api_allocine.py

```diff
@@ -51,23 +53,41 @@ def get_movie_list_page(after: str = "") -> allocine_serializers.AllocineMovieLi
     return validated_response
 
 
-def filter_invalid_edge(json_data: dict, locs: tuple) -> dict:
-    if not locs or set(locs[:2]) != {"movieShowtimeList", "edges"}:
-        # nothing to do (to complicated or locs is malformed), let it fail
-        return json_data
-
-    invalid_edge = json_data["movieShowtimeList"]["edges"].pop(locs[2])
+def _filter_invalid_edge(json_data: dict, loc: int, theater_id: str) -> dict:
+    invalid_edge = json_data["movieShowtimeList"]["edges"].pop(loc)
 
     logger.error(
         "One allocine show time edge removed (invalid)",
-        extra={"location": locs, "edge": invalid_edge},
+        extra={"theater": theater_id, "location": loc, "edge": invalid_edge},
     )
 
     return json_data
 
 
-def _parse_movie_showtimes(
-    json_data: dict, try_count: int = 0
+def _get_error_loc(err: typing.Mapping) -> int | None:
+    with suppress(KeyError, IndexError):
+        return err["loc"][2]
+    return None
+
+
+def _has_valid_error_loc(err: typing.Mapping) -> bool:
+    try:
+        loc = err["loc"]
+    except KeyError:
+        return False
+
+    try:
+        if not loc or set(loc[:2]) != {"movieShowtimeList", "edges"}:
+            # nothing to do (to complicated or loc is malformed)... ignore it
+            return False
+    except (KeyError, TypeError):
+        return False
+
+    return True
```

Comment:
Est-ce que ça ne serait pas plus simple comme ça : 

```suggestion
def _has_valid_error_loc(err: typing.Mapping) -> bool:
    try:
        is_valid_error_loc = err["loc"] and set(loc[:2]) != {"movieShowtimeList", "edges"}
    except(KeyError, TypeError):
        return False
    
    return is_valid_error_loc
```